collect_app/src/main/java/org/odk/collect/android/widgets/RatingWidget.java

```diff
@@ -14,121 +14,95 @@
+                / UiUtils.convertDpToPixel(STANDARD_WIDTH_OF_STAR, getContext()));
 
-    private void renderGrid(Context context, int numberOfStars, int columns, int rows) {
-        gridLayout.setColumnCount(columns);
-        gridLayout.setRowCount(rows);
+        binding.ratingBar1.setStepSize(1.0F);
+        binding.ratingBar2.setStepSize(1.0F);
 
-        for (int column = 0,
-             starId = 0;
-             starId < numberOfStars;
-             column++, starId++) {
-            column = column == columns ? 0 : column;
+        if (maxNumberOfStars < numberOfStars) {
+            binding.ratingBar1.setNumStars(maxNumberOfStars);
+            binding.ratingBar2.setNumStars(Math.min(numberOfStars - maxNumberOfStars, maxNumberOfStars));
 
-            ImageButton imageButton = createImageButton(context, numberOfStars, starId);
-            gridLayout.addView(imageButton);
+            binding.ratingBar2.setVisibility(View.VISIBLE);
+        } else {
+            binding.ratingBar1.setNumStars(numberOfStars);
         }
-    }
 
-    private ImageButton createImageButton(Context context, int numberOfStars, int total) {
-        ImageButton imageButton = new ImageButton(context);
-        imageButton.setImageResource(R.drawable.ic_star_border);
-        imageButton.setId(total);
-        imageButton.setPadding(0, 0, 0, 0);
-        imageButton.setBackground(null);
-        imageButton.setEnabled(!getFormEntryPrompt().isReadOnly());
-        imageButton.setOnClickListener(view -> {
-            int position = view.getId();
-            for (int i = 0; i < numberOfStars; i++) {
-                ImageButton button = (ImageButton) gridLayout.getChildAt(i);
-                button.setImageResource(i <= position ? R.drawable.ic_star : R.drawable.ic_star_border);
-            }
-            answer = position + 1;
+        binding.ratingBar1.setOnRatingBarChangeListener((ratingBar, rating, fromUser) -> {
+            binding.ratingBar1.setRating(rating);
+            binding.ratingBar2.setRating(0.0F);
+            widgetValueChanged();
+        });
 
+        binding.ratingBar2.setOnRatingBarChangeListener((ratingBar, rating, fromUser) -> {
+            binding.ratingBar1.setRating(maxNumberOfStars);
+            binding.ratingBar2.setRating(rating);
             widgetValueChanged();
         });
-        return imageButton;
+
+        binding.ratingBar1.setEnabled(!prompt.isReadOnly());
+        binding.ratingBar2.setEnabled(!prompt.isReadOnly());
+
+        if (prompt.getAnswerText() != null) {
+            int rating = Integer.parseInt(prompt.getAnswerText());
+            if (rating > maxNumberOfStars) {
+                binding.ratingBar2.setRating(rating - maxNumberOfStars);
+            } else {
+                binding.ratingBar1.setRating(rating);
+            }
+        }
+        return answerView;
     }
 
     @Override
     public void setOnLongClickListener(OnLongClickListener l) {
```

Comment:
Please fix this one-char parameter name, even simple `listener` would be better.